src/DynamoUtilities/CertificateVerification.cs

```diff
@@ -25,37 +303,19 @@ public static bool CheckAssemblyForValidCertificate(string assemblyPath)
                     "A dll file was not found at {0}. No certificate was able to be verified.", assemblyPath));
             }
 
-            //Verify that you can load the assembly into a Reflection only context
-            Assembly asm;
-            try
-            {
-                asm = Assembly.ReflectionOnlyLoadFrom(assemblyPath);
-            }
-            catch
-            {
-                throw new Exception(String.Format(
-                    "A dll file found at {0} could not be loaded.", assemblyPath));
-            }
-
-            //Verify the node library has a verified signed certificate
-            X509Certificate cert;
+            //Verify the node library file has a verified signed certificate
             try
             {
-                cert = X509Certificate.CreateFromSignedFile(assemblyPath);
+                var validCert = WinTrustInterop.WinTrust.IsTrusted(assemblyPath);
+                if (validCert)
+                {
+                    return true;
+                }
             }
             catch
             {
                 throw new Exception(String.Format(
-                    "A dll file found at {0} did not have a certificate attached.", assemblyPath));
-            }
-
-            if (cert != null)
-            {
-                var cert2 = new System.Security.Cryptography.X509Certificates.X509Certificate2(cert);
-                if (cert2.Verify())
-                {
-                    return true;
-                }
+                    "A dll file found at {0} did not have a signed certificate.", assemblyPath));
```

Comment:
Do you need to have this `throw` here?  Could you just let it fall through to the `throw` on the next line?

Reply:
The updated exceptions provide diffraction if the check failed due to exception or was actually false